packages/smooth_app/lib/pages/prices/infinite_scroll_controller.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:async';
+
+/// A generic controller for handling infinite scrolling in lists.
+/// [T] is the type of items being displayed.
+/// [P] is the type of parameters used for pagination.
+class InfiniteScrollController<T, P> {
+  InfiniteScrollController({
+    required this.fetchItems,
+    Iterable<T> initialItems = const <Never>[],
+    this.initialPage = 1,
+    this.onError,
```

Comment:
Again, we have 5 copied/pasted error managements.
Please don't make it a parameter: just always use the same code in your `catch`, with localized label `loading_dialog_default_error_message` (and the exception.toString()).